modules/geolocation/geolocation.py

```diff
@@ -129,3 +140,210 @@ def calculate_pixel_to_geo_mapping(self):
+        # note: this assumes that camera euler angles are w.r.t. plane space
+        cVector = compoundRotationMatrix.dot(cVectorCameraSpace)
+
+        # normalize since magnitude doesn't matter
+        norm = np.linalg.norm(cVector)
+        if(norm != 0) :
+            cVector = cVector / norm
+
+        cVector = np.squeeze(cVector)
+        return cVector
+
+    def __calculate_u_vector(self, compoundRotationMatrix: np.ndarray, uVectorCameraSpace=np.array([0, 1, 0])) -> np.ndarray:
+        """
+        Returns a numpy array that contains the components of the u vector (one of the camera rotation vectors)
+
+        Parameters
+        ----------
+        compoundRotationMatrix: numpy array
+            Array containing rotation matrix for camera and plane rotations
+        uVectorCameraSpace: numpy array
+            Array containing uVector in camera space (default value assumes uVector points along the pitch axis)
+            
+        Returns
+        -------
+        uVector: numpy array
+            Array containing camera rotation vector
+        """
+        # apply camera rotation to camera direction
+        # note: this assumes that camera euler angles are w.r.t. plane space
+        uVector = compoundRotationMatrix.dot(uVectorCameraSpace)
+
+        # normalize output since camera direction magnitude doesn't matter
+        norm = np.linalg.norm(uVector)
+        if (norm != 0) :
+            uVector = uVector / norm
+
+        uVector = np.squeeze(uVector)
+        return uVector
+
+    def __calculate_v_vector(self, c: np.ndarray, u: np.ndarray) -> np.ndarray:
+        """
+        Returns a numpy array that contains the components of the v vector (remaining camera rotation vector)
+
+        Parameters
+        ----------
+        c: numpy array
+            Array containing camera direction vector
+        u: numpy array
+            Array containing one camera rotation vector
+
+        Returns
+        -------
+        numpyArray:
+            array containing the remaining camera rotation vector
+        """
+        # cross product c and u to get v
+        return np.cross(c, u)
+
+    def __calculate_rotation_matrix(self, eulerAngles: dict) -> np.ndarray:
+        """
+        Calculate and return rotation matrix given euler angles
+
+        Parameters
+        ----------
+        eulerAngles: dict
+            Dictionary containing euler angles with roll ('x'), pitch ('y') and yaw ('z') rotations in radians
+
+        Returns
+        -------
+        rotationMatrix: numpy array
+            Rotation matrix calculated from the given euler angles
+        """
+        # get euler angles
+        # note: naming conventions specified in CV-Telemetry docs and commandModule specs
+        yawAngle = eulerAngles["z"]
+        pitchAngle = eulerAngles["y"]
+        rollAngle = eulerAngles["x"]
+
+        # get plane yaw rotation matrix
+        rollRotation = np.array([[1, 0, 0],
```

Comment:
Reorder yaw and pitch rotation matrices so they're in order of the matrix multiplication.